lib/ASRouter.jsm

```diff
@@ -223,6 +226,26 @@ class _ASRouter {
           newState.messages = [...newState.messages, ...messages];
         }
       }
+
+      // Some triggers have trigger listeners that we might need to initialise.
+      for (const [triggerID, listener] of ASRouterTriggerListeners) {
```

Comment:
Does it make more sense to do this the other way around... iterate over `newState.messages` and only add the listeners to the messages that have triggers? I'm assuming we don't want to add listeners if we don't even plan on showing that message. That way you wouldn't have to filter out the messages that have triggers